R/citation_test_doi.R

```diff
@@ -0,0 +1,44 @@
+#' Internal helper functions for testing 
+#' 
+#' @param library a character string of a single library source ("xdd", "plos", "scopus", "springer")
+#' @return tibble of single test doi citation known to occur in source library
+#'
+
+get_test_doi <- function(library){
+  # Load a set of example citations that should be found by the API
+  citations_file <- system.file("testdata","test-citations.csv", package="scythe")
+  citations <- read.csv(citations_file, stringsAsFactors = FALSE)
+  
+  one_citation <- citations %>% 
+    filter(grepl("^10\\.", dataone_pid), # remove non-doi identifiers
+           source == c(library)) # only need one doi
+  # select only one example
+  one_citation <- one_citation[1,] 
```

Comment:
Well I think this wraps back around to our last conversation with @jeanetteclark. If we are trying to expand the explicit nature of our testing, we would want individual tests for each `citation_search_xxx()` function and every possible search it could handle. The single DOI search was the firsts stab at expanding the testing, the next step would be to add testing for multiple DOIs (which would test the `for` loop and  `rbind()` portion of the function to combine multiple results). This is my working understanding, happy to re-visit this if that seems off.